analyze_data.py

```diff
@@ -9,6 +9,16 @@
 import sys
```

Comment:
High level feedback --

Try to exercise your code constantly and see what happens.  I didn't make it easy here, but that is something that you always have to be thinking.   You need to be constantly executing your code and seeing what comes out, and verifying it is what you expect.  This sometimes involves writing more code just as a test harness so you can know what is happening.

Without a tight feedback loop you can get into situations where multiple things are broken which makes it very hard to track down what to fix.